commit/tokenprice/outcome.go

```diff
@@ -0,0 +1,169 @@
+package tokenprice
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-ccip/shared"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+// getConsensusObservation Combine the list of observations into a single consensus observation
+func (p *processor) getConsensusObservation(
+	aos []shared.AttributedObservation[Observation],
+) (ConsensusObservation, error) {
+	aggObs := aggregateObservations(aos)
+
+	if len(aggObs.FChain) < p.bigF {
+		return ConsensusObservation{},
+			fmt.Errorf("not enough observations for FChain")
+	}
```

Comment:
Maybe this could be part of `GetConsensusMap`?

Reply:
Done